packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -62,6 +63,140 @@ const TextStyle _kToolbarButtonDisabledFontStyle = TextStyle(
 // Eyeballed value.
 const EdgeInsets _kToolbarButtonPadding = EdgeInsets.symmetric(vertical: 10.0, horizontal: 18.0);
 
+// Generates the child that's passed into CupertinoTextSelectionToolbar.
+class _CupertinoTextSelectionToolbarWrapper extends StatefulWidget {
+  const _CupertinoTextSelectionToolbarWrapper({
+    Key key,
+    this.barTopY,
+    this.arrowTipX,
+    this.handleCut,
+    this.handleCopy,
+    this.handlePaste,
+    this.handleSelectAll,
+    this.isArrowPointingDown,
+  }) : super(key: key);
+
+  final double arrowTipX;
+  final double barTopY;
+  final VoidCallback handleCut;
+  final VoidCallback handleCopy;
+  final VoidCallback handlePaste;
+  final VoidCallback handleSelectAll;
+  final bool isArrowPointingDown;
+
+  @override
+  _CupertinoTextSelectionToolbarWrapperState createState() => _CupertinoTextSelectionToolbarWrapperState();
+}
+
+class _CupertinoTextSelectionToolbarWrapperState extends State<_CupertinoTextSelectionToolbarWrapper> with WidgetsBindingObserver {
+  // Whether the user's system clipboard is empty as of the last check.
+  bool _isClipboardEmpty;
+
+  // Asynchronously check if theh clipboard has data and update
+  // _isClipboardEmpty if needed.
+  void _updateClipboardStatus() {
+    Clipboard.getData(Clipboard.kTextPlain).then((ClipboardData value) {
+      setState(() {
+        if (!mounted) {
+          return;
+        }
+        final bool isClipboardEmpty = value?.text == null || value.text.isEmpty;
+        if (isClipboardEmpty == _isClipboardEmpty) {
+          return;
+        }
+        _isClipboardEmpty = isClipboardEmpty;
```

Comment:
move everything except for this line outside of the setState? (Only this line is actually setting state).

Reply:
I moved everything inside because I was getting weird test failures when the widget would unmount, but I don't see anything like that now.  I'll move it outside.